packages/graphql-yoga/src/plugins/types.ts

```diff
@@ -40,51 +39,15 @@ export interface OnRequestEventPayload<TServerContext> {
   endResponse(response: Response): void
 }
 
-export type OnRequestParseHook = (
-  payload: OnRequestParseEventPayload,
-) => PromiseOrValue<void | OnRequestParseHookResult>
-
-export type RequestParser = (request: Request) => PromiseOrValue<GraphQLParams>
-
-export interface OnRequestParseEventPayload {
-  request: Request
-  requestParser: RequestParser | undefined
-  setRequestParser: (parser: RequestParser) => void
-}
-
-export type OnRequestParseHookResult = {
-  onRequestParseDone?: OnRequestParseDoneHook
-}
-
-export type OnRequestParseDoneHook = (
-  payload: OnRequestParseDoneEventPayload,
+export type OnPrepareHook = (
+  payload: OnPrepareEventPayload,
 ) => PromiseOrValue<void>
 
-export interface OnRequestParseDoneEventPayload {
+export interface OnPrepareEventPayload {
```

Comment:
onRequestParse and onRequestParseDone hooks are seperated because subsequent plugins can change the way of parsing before parsing has been done by Yoga's request parsers, so `requestParser` isn't called in here yet.
 `onRequestParseDone` is called after the parameters are parsed in order to let the user change the parsed params for the execution. For example, persisted operations use this to manipulate the parameters for the following execution.

Reply:
I understand that, my idea was to not have the user control the parsing. Yoga should implement all official and unofficial parsing techniques and do the parsing for the user, and have him manipulate the results only.

But this is debatable...